File: EmployeeAdminPortal/EmployeeAdminPortal/Models/Entities/LeaveRequest.cs
namespace EmployeeAdminPortal.Models.Entities
{
    public class LeaveRequest
    {
        public Guid Id { get; set; }
        public Guid EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Reason { get; set; }
        public LeaveStatus Status { get; set; } = LeaveStatus.Pending;
        public Guid? ApprovedByUserId { get; set; }
        public User? ApprovedByUser { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: client/src/types/leaveRequest.ts
export type LeaveStatus = "Pending" | "Approved" | "Rejected";

export interface LeaveRequest {
  id: string;
  employeeId: string;
  startDate: string;
  endDate: string;
  reason?: string | null;
  status: LeaveStatus;
  approvedByUserId?: string | null;
  createdAt: string;
}

export interface AddLeaveRequestDto {
  employeeId: string;
  startDate: string;
  endDate: string;
  reason?: string;
}
